### PR TITLE
Viber

### PR DESCRIPTION
**Pull request by nedrichards**
Extra data based - so works for x86-64 only.

Only issue is that it seems to break the compositor slightly the first time it logs in (I suspect it's messing with its window class). Works fine after you restart gnome shell or close and reopen the Viber app.

### CODE REVIEW

File: com.viber.Viber.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "separate-locales": false,
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=home",
+    /* needed for video chat */
+    "--device=all",
+    "--talk-name=org.freedesktop.Notifications",
```
Comment (TingPing) [inline com.viber.Viber.json]: Does this have a tray icon? If so you'll need the appindicator bits (see discord flatpak).

File: com.viber.Viber.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "viber",
+  "separate-locales": false,
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=home",
```
Comment (TingPing) [inline com.viber.Viber.json]: Does it really need full access to home?

File: com.viber.Viber.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+              "rm -f viber.deb",
+              "tar xf data.tar.xz",
+              "rm -f control.tar.gz data.tar.xz debian-binary",
+              "mv opt/* .",
+              "mv usr/* .",
+              "chmod -R a-s,go+rX,go-w .",
+              "rmdir usr",
+              "mkdir -p export/share/applications",
+              "sed 's|Icon=/usr/share/pixmaps/viber.png|Icon=com.viber.Viber|; s|Exec=/usr/bin/viber|Exec=viber|' share/applications/viber.desktop > export/share/applications/com.viber.Viber.desktop",
+              "echo StartupWMClass=Viber >> export/share/applications/com.viber.Viber.desktop",
```
Comment (TingPing) [inline com.viber.Viber.json]: `desktop-file-edit --set-key=StartupWMClass --set-value=Viber export/share/applications/com.viber.Viber.desktop`

is probably more robust.

File: com.viber.Viber.json (hunk 2 above)
Comment (nedrichards) [reply com.viber.Viber.json]: It certainly needs access to ~/.Viber-PC to save it's config and it needs the ability to upload and save arbitrary files. I'd be happy to take suggestions for a smaller scope.

File: com.viber.Viber.json (hunk 1 above)
Comment (nedrichards) [reply com.viber.Viber.json]: Yes, this does. Will add that.

File: com.viber.Viber.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=home",
+    /* needed for video chat */
+    "--device=all",
+    "--talk-name=org.freedesktop.Notifications",
+    "--extra-data=viber.deb:bacc38034fa6d926e25122c152c867a4f576b59e920af58ce904c4e9f255773f:84861076::http://download.cdn.viber.com/cdn/desktop/Linux/viber.deb"
```
[the remark this answers is not in the document]
Comment (nedrichards) [reply com.viber.Viber.json]: I'm prepared for this to break ;-)

File: com.viber.Viber.json (hunk 3 above)
Comment (nedrichards) [reply com.viber.Viber.json]: thanks

File: com.viber.Viber.json (hunk 2 above)
Comment (TingPing) [reply com.viber.Viber.json]: Well you might be able to do something like override `$HOME` so it at least saves the config into its private directory. Keeping the access outside of that is probably fine though.

**Comment by cosimoc** [comment]
Test build: https://flathub.org/builds/#/builders/1/builds/535

File: com.viber.Viber.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  "runtime-version": "1.6",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "viber",
+  "separate-locales": false,
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=pulseaudio",
+    "--share=network",
+    "--filesystem=~/.Viber-PC",
```
Comment (TingPing) [inline com.viber.Viber.json]: I made this comment before but what did you think about maybe setting `$HOME` to `$XDG_CONFIG_HOME` or such so we keep configuration contained.

Comment (nedrichards) [reply com.viber.Viber.json]: That seems like a good idea.

Comment (nedrichards) [reply com.viber.Viber.json]: How were you envisaging doing that? Either I'm doing it wrong (not unexpected) or that doesn't work as well as I'd hoped.

Comment (TingPing) [reply com.viber.Viber.json]: Eh don't worry about it I guess.

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/com.viber.Viber

**Pull request closed without merging**